src/app.rs

```diff
@@ -1842,8 +1847,14 @@ impl Application for App {
                 if let Some(tab) = self.tab_model.data::<Tab>(entity) {
                     self.activate_nav_model_location(&tab.location.clone());
                 }
-
-                return self.update_title();
+                let mut commands = vec![];
+                commands.push(self.update_title());
+                // if the tab was in an active search mode
```

Comment:
TabActivate happens when a tab is clicked on. I'm not sure why this needs to rerun the search.

Reply:
There is the issue when multiple search tabs are open (with the same search input or slightly different that could have a joint set of search result), so what i did since we do not have interactive search as gnome-files is to research when you activate a tab that had a search active.

Reply:
This is good enough for now, I suppose.